packages/dev/core/src/Shaders/ShadersInclude/pbrBlockReflection.fx

```diff
@@ -264,7 +267,12 @@
             #else
                 vec3 irradianceVector = vec3(reflectionMatrix * vec4(normalW, 0)).xyz;
             #endif
-            
+            vec3 irradianceView = vec3(reflectionMatrix * vec4(viewDirectionW, 0)).xyz;
+            #if !defined(USE_IRRADIANCE_DOMINANT_DIRECTION) && !defined(REALTIME_FILTERING)
+                float NdotV = max(dot(normalW, viewDirectionW), 0.0);
+                irradianceVector = mix(irradianceVector, irradianceView, (0.5 * (1.0 - NdotV)) * diffuseRoughness);

```

Comment:
Doesn't this calculation depend on the type of the diffuse BRDF we are currently using? In other words, will Burley have a different calculation from EON (Lambert is view independent, so we shouldn't execute this code in that case)?

Also, this is a breaking change in Burley case (our default diffuse), I don't know if we should do something about it? cc @sebavan

Reply:
So, this is the bent normal hack to approximate diffuse roughness for IBL's if you aren't doing realtime filtering or prefiltering with CDF. This works for SH as well and I don't have a different version for Burley or EON. They use the same logic.

Yes, I suppose we could disable this if the user has chosen Lambert as the diffuse roughness mode. That's basically the same as disabling diffuse roughness.